akd/src/append_only_zks.rs

```diff
@@ -319,10 +319,11 @@ impl Azks {
         let mut longest_prefix_children = [Node {
             label: EMPTY_LABEL,
             hash: crate::utils::empty_node_hash(),
-        }; ARITY];
+        }; ARITY as usize];
         for i in 0..ARITY {
+            let dir = Direction::try_from(i).unwrap();
```

Comment:
This will be an error since Direction::try_from(i).unwrap() when i=0 will yield Direction::None, whereas we mean for it to yield Direction::Left.

Perhaps we can just do:
```
for dir in [Direction::Left, Direction::Right] {
  ...
}
```
and avoid the unwrap() as well.

This doesn't respect the ARITY = 2 requirement, but... perhaps we should also get rid of ARITY as well.

Also, I'm surprised that our tests didn't catch this error. Any idea why that might be happening?

Reply:
`Direction::Left` is `0u8` so it's valid for a Direction. It's why we map `Direction::None` to `u8::MAX`

Reply:
So I've removed the loops over ARITY (in upcoming update), but maybe it's helpful to have ARITY stay around since we have some array initializers which point to that const. Even if just treated as a name alias, it removes the mystical `[...;2]` in array inits around the lib?

Anyways the direction array definitely helps clean things up!

Reply:
Ohh, I see, good to know that it was my mistake in interpreting it as a bug when it was actually working correctly, and not a lapse in our tests, either. Thanks!